versioned_docs/version-V3/guides/providing-liquidity/setting-up-your-contract.md

```diff
@@ -0,0 +1,188 @@
+---
+id: setting-up
+title: Setting Up Your Contract
+sidebar_position: 1
+---
+
+## Setting up the Contract
+
+First we declare the solidity version that will be used to compile the contract, and the `abicoder v2` to allow arbitrary nested arrays and structs to be encoded and decoded in calldata, a feature we use when transacting with a pool.
+
+```solidity
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity =0.7.6;
+pragma abicoder v2;
+```
+
+Next, import the two contracts we'll be using from the npm package installation
+
+```solidity
+import '@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol';
+import '@uniswap/v3-core/contracts/libraries/TickMath.sol';
+import '@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol';
+import '@uniswap/v3-periphery/contracts/interfaces/ISwapRouter.sol';
+import '@uniswap/v3-periphery/contracts/interfaces/INonfungiblePositionManager.sol';
+import '@uniswap/v3-periphery/contracts/libraries/TransferHelper.sol';
+import '@uniswap/v3-periphery/contracts//base/LiquidityManagement.sol';
+```
+
+Here we create our contract called `LiquidityExamples`, and inherit both `IERC721Receiver` and `LiquidityManagement`.
+
+Here we hardcode the token contract addresses and pool fee tiers for our example. In production, you would likely use in input parameter for this and pass the input into a memory variable, allowing you to change what the pools and tokens you are interacting with on a per transaction basis, but for conceptual simplicity we are hardcoding them here.
+
+```solidity
+contract LiquidityExamples is IERC721Receiver, LiquidityManagement {
+
+    address public constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address public constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+
+    uint24 public constant poolFee = 3000;
+```
+
+We declare an immutable public variable `nonfungiblePositionManager` of type `INonfungiblePositionManager`:
+
+```solidity
+    INonfungiblePositionManager public immutable nonfungiblePositionManager;
+```
+
+## Allowing ERC721 Interactions
+
+Every [NFT](https://eips.ethereum.org/EIPS/eip-721) is identified by a unique uint256 ID inside the ERC-721 smart contract, declared as the `tokenId`
+
+To allow deposits of ERC721 expressions of liquidity, we'll need to create a struct called `Deposit`, a mapping of `uint256` to the `Deposit` struct, then declare that mapping as a public variable `deposits`.
+
+```solidity
+    struct Deposit {
+        address owner;
+        uint128 liquidity;
+        address token0;
+        address token1;
+    }
+
+    mapping(uint256 => Deposit) public deposits;
+```
+
+```solidity
```

Comment:
The constructor is a little different now since we are not inheriting LiquidityManagement (see new code pushed)

Reply:
Also maybe we can delete this one since we go over the constructor below?